### PR TITLE
Feature/hyperledger po

### PR DESCRIPTION
**Pull request by matthias-geuchen**
### Reason for this PR
- to validate fields of study specified in matriculation data and do basic module registration in chaincode, we need to store simple examination regulations (Prüfungsordnungen) on chain

### Changes in this PR
- add basic examination regulation chaincode

### CODE REVIEW

**Comment by bastihav** [comment]
Please see #68 for the api models.. in short:
examination regulations has a name and a list of modules, modules have an id (M.0000.00000) and name (Algorithms).

Let me know if you would like to discuss those objects further.

I will review when the APIs are consistent

File: hlf/chaincode/examination-regulation.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+In the following, the chaincode api for the "ExaminationRegulation"-Chaincode is described.
+This contains all Transactions and Models provided for this domain.
+The Errors returned are defined [here](errors.md#Errors).
+
+## Transactions
+
+### AddExaminationRegulation
+- ID = addExaminationRegulation
+- Send
+    - name :: String ,examinationRegulation :: List\<String\>
```
Comment (NikoBergemann) [inline hlf/chaincode/examination-regulation.md]:
```suggestion
    - name :: String 
    - examinationRegulation :: List\<String\>

```

Comment (NikoBergemann) [inline hlf/chaincode/examination-regulation.md]:
```suggestion
    - name :: String ,examinationRegulation :: List\<String\>
    - isOpen :: Boolean

```
can new students still enroll in that course?

File: hlf/chaincode/examination-regulation.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - [GenericError](errors.md#GenericError) 
+      ```json
+      {
+        "type": "HLUnprocessableLedgerState",
+        "title": "The state on the ledger does not conform to the specified format"
+      }
+      ```
+      - Description: This error is returned, if the state of data on the ledger is not consistent with the current model. This error should only occur if the model changes while the old ledger state remains without modification.
+
+
```
Comment (NikoBergemann) [inline hlf/chaincode/examination-regulation.md]:
```suggestion
### CloseExaminationRegulation
- ID = closeExaminationRegulation
- Send
    - name :: String
- Receive
    - isOpen :: Boolean
      - Description: new "isOpen"-state
      
    - [GenericError](errors.md#GenericError) 
      ```json
      {
        "type": "HLNotFound",
        "title": "There is no examination regulation for the given name"
      }
      ```
      - Description: This error is returned, if the examination regulation for the name is not present on the ledger.
    - [GenericError](errors.md#GenericError) 
      ```json
      {
        "type": "HLUnprocessableLedgerState",
        "title": "The state on the ledger does not conform to the specified format"
      }
      ```
      - Description: This error is returned, if the state of data on the ledger is not consistent with the current model. This error should only occur if the model changes while the old ledger state remains without modification.

```

**Review by NikoBergemann — CHANGES_REQUESTED** [review]

**Comment by NikoBergemann** [comment]
> Please see #68 for the api models.. in short:
> examination regulations has a name and a list of modules, modules have an id (M.0000.00000) and name (Algorithms).
> 
> Let me know if you would like to discuss those objects further

I think this is a great idea, Unique Identifiers are always a useful thing

**Review by dhelmert — CHANGES_REQUESTED** [review]
In the lagom-vue api, we defined an endpoint to get the names of all examination regulations.
Even though we intend to cache the regulations at lagom, an endpoint to fetch all regulations will probybly be useful.
For performance reasons, we want to limit the number of calls to hyperledger as much as possible. Therefore, having the "GET" transaction take a list of list of names and returning a list of regulations might be more useful.

**Comment by dhelmert** [comment]
> Please see #68 for the api models.. in short:
> examination regulations has a name and a list of modules, modules have an id (M.0000.00000) and name (Algorithms).
> 
> Let me know if you would like to discuss those objects further.
> 
> I will review when the APIs are consistent

Also, the examination regulation object now has an "active" flag

File: hlf/chaincode/examination-regulation.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+### CloseExaminationRegulation
```
Comment (bastihav) [inline hlf/chaincode/examination-regulation.md:61]: Do we need an error if the examination regulation was already closed?

**Review by bastihav — APPROVED** [review]

**Review by dhelmert — APPROVED** [review]

File: hlf/chaincode/examination-regulation.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+### <a id="ExaminationRegulation">ExaminationRegulation
+```json
+{
+  "name": "Computer Science v3",
+  "active": true,
+  "modules": [
+    {
+      "id": "M.1275.01158",
+      "name": "Math 1"
```
Comment (NikoBergemann) [inline hlf/chaincode/examination-regulation.md:96]:
```suggestion
      module: [Module](#Module)

```
DRY

**Review by NikoBergemann — CHANGES_REQUESTED** [review]
love the new GetExaminationRegulations - Logic

File: hlf/chaincode/examination-regulation.md (hunk 4 above)
Comment (matthias-geuchen) [reply hlf/chaincode/examination-regulation.md:96]: True, however we decided to have valid json examples in the api documents back when we specified the matriculation api. Hence we will keep this (for now).

**Review by NikoBergemann — APPROVED** [review]

**Review by ldklenner — APPROVED** [review]

**Pull request merged**